src/test/java/at/ac/tuwien/kr/alpha/solver/RegressionTest.java

```diff
@@ -0,0 +1,17 @@
+package at.ac.tuwien.kr.alpha.solver;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@ParameterizedTest
+@MethodSource("at.ac.tuwien.kr.alpha.solver.RegressionTestConfigProvider#provideConfigs")
+public @interface RegressionTest {
```

Comment:
I would like to have a discussion about the term "regression test", because a regression --- at least to my understanding --- is a re-introduced and re-appearing bug that was already solved. Some of our tests come from bugs, so the term regression test would be suitable for them. However, most of our tests are more generally testing some functionality. So, I think the more general term of `FunctionalTest` is more fitting here (also see https://en.wikipedia.org/wiki/Functional_testing which describes regression testing as one kind of functional testing).

So, my opinion is that most stuff that is now named some kind of "regression" test would better be named "functional" test. We could then have `FunctionalTest`, `FunctionalTestConfig`, `FunctionalTestConfigProvider` and `@FunctionalTest` as annotation.
Note that some stuff might still be named with regression, namely if we really want to indicate what test cases come from regressions. An example of such would be the `DefaultSolverTest`.